packages/core/src/external/carequality/ihe-gateway-v2/outbound/xca/process/dr-response.ts

```diff
@@ -90,67 +92,71 @@ async function parseDocumentReference({
+    if (entry.docUniqueId && entry.metriportId) {
+      acc[stripUrnPrefix(entry.docUniqueId)] = entry.metriportId;
+    }
+    return acc;
+  }, {});
+}
+
 async function handleSuccessResponse({
   documentResponses,
   outboundRequest,
   gateway,
+  s3Utils,
 }: {
-  //eslint-disable-next-line @typescript-eslint/no-explicit-any
-  documentResponses: any;
+  documentResponses: DocumentResponse[];
   outboundRequest: OutboundDocumentRetrievalReq;
   gateway: XCAGateway;
+  s3Utils: S3Utils;
 }): Promise<OutboundDocumentRetrievalResp> {
-  const s3Utils = new S3Utils(region);
-
-  const idMapping = outboundRequest.documentReference.reduce(
-    (acc: Record<string, string>, entry) => {
-      if (entry.docUniqueId && entry.metriportId) {
-        acc[stripUrnPrefix(entry.docUniqueId)] = entry.metriportId;
-      }
-      return acc;
-    },
-    {}
-  );
-
-  //eslint-disable-next-line @typescript-eslint/no-explicit-any
-  const documentReferences = Array.isArray(documentResponses)
-    ? await Promise.all(
-        documentResponses.map(async (documentResponse: DocumentResponse) =>
-          parseDocumentReference({ documentResponse, outboundRequest, s3Utils, idMapping })
+  try {
+    const idMapping = generateIdMapping(outboundRequest.documentReference);
+    const documentReferences = Array.isArray(documentResponses)
+      ? await Promise.all(
+          documentResponses.map(async (documentResponse: DocumentResponse) =>
+            parseDocumentReference({ documentResponse, outboundRequest, s3Utils, idMapping })
+          )
         )
-      )
-    : [
-        await parseDocumentReference({
-          documentResponse: documentResponses,
-          outboundRequest,
-          s3Utils,
-          idMapping,
-        }),
-      ];
+      : [
+          await parseDocumentReference({
+            documentResponse: documentResponses,
+            outboundRequest,
+            s3Utils,
+            idMapping,
+          }),
+        ];
 
-  const response: OutboundDocumentRetrievalResp = {
-    id: outboundRequest.id,
-    patientId: outboundRequest.patientId,
-    timestamp: outboundRequest.timestamp,
-    responseTimestamp: dayjs().toISOString(),
-    gateway,
-    documentReference: documentReferences,
-  };
-  return response;
+    const response: OutboundDocumentRetrievalResp = {
+      id: outboundRequest.id,
+      patientId: outboundRequest.patientId,
+      timestamp: outboundRequest.timestamp,
+      responseTimestamp: dayjs().toISOString(),
+      gateway,
+      documentReference: documentReferences,
+    };
+    return response;
+  } catch (error) {
+    throw new MetriportError(`Error Processing Success Response`, error);
+  }
 }
 
-export async function processDRResponse({
+export async function processDRResponseSOAP({
   drResponse: { response, success, gateway, outboundRequest },
+  s3Utils,
 }: {
   drResponse: DRSamlClientResponse;
+  s3Utils: S3Utils;
```

Comment:
Why are we adding S3Utils to these function signatures, passing it over multiple layers? Is it just to help with the script at `packages/utils`?

For testing purposes, we can mock a class w/o having to pass it as parameter.

If we want to make it easier for `packages/utils` to send a custom implementation of S3Utils, we could have a function to return the S3Utils on the specific code that needs it, and then override it with something like [this](https://www.geeksforgeeks.org/how-to-override-functions-of-module-in-node-js/).

My point is that ideally the code is built to what makes sense to that particular piece of code. As we add more stuff than it needs/does, it gets more complicated, which makes it harder to maintain and more likely to break.